src/main/kotlin/com/example/petshopkotlin/user/UserService.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.petshopkotlin.user
+
+import com.example.petshopkotlin.user.model.Role
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
+import org.springframework.stereotype.Service
+import com.example.petshopkotlin.user.model.User as ModelUser
+
+@Service
+class UserService(
+    @Autowired val userRepo: UserRepository,
+) : UserDetailsService {
+    fun createUser(user: ModelUser): String {
+        require(!userRepo.existsByUserName(user.userName)) { "User with ${user.userName} already exists!" }
+        val errors = validateUser(user)
+        require(errors.isBlank()) { errors }
+
+        val password = encryptPassword(user.password)
```

Comment:
Server side encryption is good, however there is an important consideration when choosing server side encryption, can you think of it?

Reply:
Should I have a strong algorithm for encryption?